osu.Framework/Timing/InterpolatingFramedClock.cs

```diff
@@ -7,106 +7,117 @@ namespace osu.Framework.Timing
+            realtimeClock.ProcessFrame();
             FramedSourceClock.ProcessFrame();
 
             sourceIsRunning = FramedSourceClock.IsRunning;
 
-            LastInterpolatedTime = currentTime;
+            lastInterpolatedTime = currentTime;
 
             if (FramedSourceClock.IsRunning)
             {
                 if (FramedSourceClock.ElapsedFrameTime != 0)
                     IsInterpolating = true;
 
-                CurrentInterpolatedTime += clock.ElapsedFrameTime * Rate;
+                currentInterpolatedTime += realtimeClock.ElapsedFrameTime * Rate;
 
-                if (!IsInterpolating || Math.Abs(FramedSourceClock.CurrentTime - CurrentInterpolatedTime) > AllowableErrorMilliseconds)
+                if (!IsInterpolating || Math.Abs(FramedSourceClock.CurrentTime - currentInterpolatedTime) > AllowableErrorMilliseconds)
                 {
                     // if we've exceeded the allowable error, we should use the source clock's time value.
                     // seeking backwards should only be allowed if the source is explicitly doing that.
-                    CurrentInterpolatedTime = FramedSourceClock.ElapsedFrameTime < 0 ? FramedSourceClock.CurrentTime : Math.Max(LastInterpolatedTime, FramedSourceClock.CurrentTime);
+                    currentInterpolatedTime = FramedSourceClock.ElapsedFrameTime < 0 ? FramedSourceClock.CurrentTime : Math.Max(lastInterpolatedTime, FramedSourceClock.CurrentTime);
 
                     // once interpolation fails, we don't want to resume interpolating until the source clock starts to move again.
                     IsInterpolating = false;
                 }
                 else
                 {
                     //if we differ from the elapsed time of the source, let's adjust for the difference.
-                    CurrentInterpolatedTime += (FramedSourceClock.CurrentTime - CurrentInterpolatedTime) / 8;
+                    currentInterpolatedTime += (FramedSourceClock.CurrentTime - currentInterpolatedTime) / 8;
 
                     // limit the direction of travel to avoid seeking against the flow.
-                    CurrentInterpolatedTime = Rate >= 0 ? Math.Max(LastInterpolatedTime, CurrentInterpolatedTime) : Math.Min(LastInterpolatedTime, CurrentInterpolatedTime);
+                    currentInterpolatedTime = Rate >= 0 ? Math.Max(lastInterpolatedTime, currentInterpolatedTime) : Math.Min(lastInterpolatedTime, currentInterpolatedTime);
                 }
             }
 
-            currentTime = sourceIsRunning ? CurrentInterpolatedTime : FramedSourceClock.CurrentTime;
+            currentTime = sourceIsRunning ? currentInterpolatedTime : FramedSourceClock.CurrentTime;
         }
+
+        private void resetInterpolation()
+        {
+            currentTime = 0;
+            lastInterpolatedTime = 0;
+            currentInterpolatedTime = 0;
+        }
+
+        double IFrameBasedClock.FramesPerSecond => throw new NotImplementedException();
```

Comment:
Not 100% on this either.

- `NotSupportedException` may be better fit than `NotImplementedException`
- Used to be previously 0 rather than throwing. This probably doesn't matter, but `FramedClock.ToString()` is using it, which may turn out rather nasty in contexts like debugging wherein debugger will just use `.ToString()` for watch output or something and will instead start throwing exceptions.

Reply:
I'm fine with either. I was also hesitant due to the `ToString()` usage, but one would hope there's no way that can hit this usage, because `FrameBasedClock` does its own FPS calculation, and `Interpolated..`  doesn't derive from it.

Reply:
Let's maybe do 0 then for now to be on the safe side.

```suggestion
        double IFrameBasedClock.FramesPerSecond => 0;
```